doc/developer_guide.md

```diff
@@ -0,0 +1,34 @@
+---
+title: "Developer guide"
+date: 2021-11-18
+draft: false
+weight: 30
+description: "Setting up for KCIDB development"
+---
+Hacking
+-------
+
+If you want to hack on the source code, install the package in the editable
+mode with the `-e/--editable` option, and with "dev" extra included. E.g.:
+
+    pip3 install --user --editable '.[dev]'
+
+The latter installs kcidb executables which use the modules from the source
+directory, and changes to them will be reflected immediately without the need
+to reinstall. It also installs extra development tools, such as `flake8` and
+`pylint`.
+
+Then make sure your PATH includes the `~/.local/bin` directory, e.g. with:
+
+    export PATH="$PATH":~/.local/bin
+
+Releasing
+---------
+
+Before releasing make sure the README.md and SUBMISSION_HOWTO.md are up to
```

Comment:
There's no `SUBMISSION_HOWTO.md` file any more, there's the submitter guide instead.

There is still a `README.md` but I'm not sure if other parts of the docs should be mentioned here now that things have been reorganised.

Reply:
Good catch, changed to just "documentation files", since there's more of them now, anyway.